src/auth/ha-auth-flow.js

```diff
@@ -153,6 +153,30 @@ class HaAuthFlow extends localizeLiteMixin(PolymerElement) {
     }
 
     this.setProperties(props);
+    if (step.step_id === "mfa") {
+      this._processMfa(step);
+    }
+  }
+
+  async _processMfa(step) {
+    const placeholders = step.description_placeholders || {};
+    const data = step.data || {};
+
+    switch (placeholders.mfa_module_id) {
```

Comment:
I'm not a fan of passing data to the frontend like this. We should find a better approach.